device/sycl/src/clusterization/clusterization_algorithm.sycl

```diff
@@ -27,11 +28,24 @@ class ccl_kernel;
 
 clusterization_algorithm::clusterization_algorithm(
     const traccc::memory_resource& mr, vecmem::copy& copy, queue_wrapper queue,
-    const unsigned short target_cells_per_partition)
+    const config_type& config)
     : m_mr(mr),
       m_queue(queue),
       m_copy(copy),
-      m_target_cells_per_partition(target_cells_per_partition) {}
+      m_config(config),
+      m_f_backup(m_config.backup_size(), m_mr.main),
+      m_gf_backup(m_config.backup_size(), m_mr.main),
+      m_adjc_backup(m_config.backup_size(), m_mr.main),
+      m_adjv_backup(m_config.backup_size() * 8, m_mr.main),
+      m_backup_mutex(vecmem::make_unique_alloc<unsigned int>(m_mr.main)) {
+    unsigned int zero = 0;
+    details::get_queue(m_queue)
+        .memcpy(
+            m_backup_mutex.get(), &zero,
+            sizeof(
+                std::remove_extent_t<decltype(m_backup_mutex)::element_type>))
+        .wait_and_throw();
```

Comment:
You could use `cl::sycl::queue::memset`. It's only in very old SYCL compilers that it's not available. Which we shouldn't worry about with this project. (In VecMem we have an explicit test for whether that call is available or not.)

And unfortunately you are correct to use `wait_and_throw()` here. :frowning: Unlike CUDA, here we **do** need to wait for this operation to finish.

Finally: You forgot about the `vecmem::copy::setup` calls for all of the buffers. (Yes, wait for those as well...)